packages/fxa-auth-db-mysql/db-server/test/backend/db_tests.js

```diff
@@ -2597,6 +2597,84 @@ module.exports = function (config, DB) {
       })
     })
 
+    describe('account subscriptions', () => {
+      const subs = [...Array(10)].map(() => 'sub-' + Math.random())
+      let account
+
+      beforeEach(() => {
+        account = createAccount()
+        return db.createAccount(account.uid, account)
+      })
+
+      it('should create a new subscription for account', () => {
+        return db.createAccountSubscription(account.uid, subs[0], 'prod0', Date.now())
+          .then(result => assert.deepEqual(result, {}))
+      })
+
+      it('should fail to create for unknown account', () => {
+        return db.createAccountSubscription('12312312312', subs[1], 'prod2', Date.now())
+          .then(assert.fail, err => assert.equal(err.errno, 116, 'not found'))
+      })
+
+      it('should fail to create duplicate subscription id', () => {
+        const account2 = createAccount()
+        return db.createAccount(account2.uid, account2)
+          .then(() => db.createAccountSubscription(account.uid, subs[2], 'prod2', Date.now()))
+          .then(() => db.createAccountSubscription(account2.uid, subs[2], 'prod3', Date.now()))
+          .then(assert.fail, err => assert.equal(err.errno, 101, 'conflict'))
+      })
+
+      const pickSet = (list, name) => new Set(list.map(x => x[name]))
+
+      it('should support fetching all subscriptions', () =>
+        Promise.all([
```

Comment:
how come these tests do not `return` the promises?

Reply:
They do, at least implicitly with the arrow function

Reply:
ah i see it now, didn't see the full structure of the `it()` 👍